### PR TITLE
Security: integrate pam_mount into PAM of NixOS

### PR DESCRIPTION
**Pull request by ts468**

### CODE REVIEW

**Comment by Fuuzetsu** [comment]
@vcunat @edolstra @lethalman seem to be the people to look at this

File: nixos/modules/security/pam_mount.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          xlink:href="http://pam-mount.sourceforge.net/pam_mount.conf.5.html" />.
+        '';
+      };
+    };
+
+  };
+
+  config = mkIf (cfg.enable || anyPamMount) {
+
+    #environment.systemPackages = [ pkgs.pam_mount pkgs.utillinux ];
```
Comment (lethalman) [inline nixos/modules/security/pam_mount.nix]: Remove this commented line.

File: nixos/modules/services/x11/display-managers/slim.nix [hunk 2]
```diff
@@ -128,7 +128,7 @@ in
 
     # Allow null passwords so that the user can login as root on the
     # installation CD.
-    security.pam.services.slim = { allowNullPassword = true; startSession = true; };
+    security.pam.services.slim = { allowNullPassword = true; startSession = true; pamMount = true; };
```
Comment (lethalman) [inline nixos/modules/services/x11/display-managers/slim.nix]: Why do you force it to true?

**Comment by lethalman** [comment]
I'm wondering, how does the user specify the password during login?

File: nixos/modules/services/x11/display-managers/slim.nix (hunk 2 above)
Comment (ts468) [inline nixos/modules/services/x11/display-managers/slim.nix]: What would be the proper way to enable pam_mount for, e.g., slim instead?

**Comment by ts468** [comment]
The user login password is taken for opening, e.g., a LUKS device mapper. The device mapper would be expected to be created with the corresponding user login password. That is, no separate password has to be specified.

**Comment by ts468** [comment]
PR #8636 is a prerequisite for pam_mount to work. So could you merge #8636 as well, please?

File: nixos/modules/services/x11/display-managers/slim.nix (hunk 2 above)
Comment (lethalman) [inline nixos/modules/services/x11/display-managers/slim.nix]: Isn't pamMount enabled when security.pam.mount.enable is enabled? That true is forcing it, isn' it?

**Comment by lethalman** [comment]
@ts468 I mean, where is the password prompt displayed? Given that it will be done during X login, will the password be asked by some X tool?

File: nixos/modules/services/x11/display-managers/slim.nix (hunk 2 above)
Comment (ts468) [inline nixos/modules/services/x11/display-managers/slim.nix]: True, thanks!

**Comment by ts468** [comment]
@lethalman, there is no extra password prompt, at least as far as I can tell. The password is passed on from, e.g., the authentication module of the PAM system.

**Comment by lethalman** [comment]
@ts468 ah ok... was just curiosity

**Comment by lethalman** [comment]
Looks fine to me as long as it works for you, thanks.

**Comment by obadz** [comment]
@ts468,

Nice to see that you got pam_mount working! I was playing with it a while back (the branch containing my failed work is here: https://github.com/obadz/nixpkgs/tree/pam_mount) but I gave up when I eventually got ecryptfs to work.

Would you mind explaining briefly what the procedure is to configure pam_mount on NixOS, maybe in the NixOS wiki near where ecryptfs is mentioned? (https://nixos.org/wiki/Encrypted_Root_on_NixOS#Encrypted_Individual_Home_Directories_with_eCryptfs)

**Comment by ts468** [comment]
@obadz, thanks, I'll look into it

**Pull request merged**